TLM/TMPE.CitiesGameBridge/Service/NetService.cs

```diff
@@ -102,71 +102,108 @@ public class NetService : INetService {
             IterateNodeSegments(nodeId, ClockDirection.None, handler);
         }
 
-        public void IterateNodeSegments(ushort nodeId,
-                                        ClockDirection dir,
-                                        NetSegmentHandler handler) {
-            NetManager netManager = Singleton<NetManager>.instance;
+        /// <summary>
+        /// Given a <paramref name="nodeId"/>, retrieves the assocaited <see cref="NetNode"/>.
+        ///
+        /// A <c>nodeId</c> of <c>0</c> returns <c>false</c>.
+        /// </summary>
+        /// 
+        /// <param name="nodeId">The id of the segment to retrieve.</param>
+        /// <param name="node">The returned segment.</param>
+        /// 
+        /// <returns>Returns <c>true</c> if successful, otherwise <c>false</c>.</returns>
+        public bool TryGetNode(ushort nodeId, out NetNode node) {
+            if (nodeId == 0) {
```

Comment:
shouldn't we check if the created flag is set and delete flag is not? (nodeId==0 does not have the created flag )

Reply:
Well, this is why I was asking for more input. I would assume that would be a good thing to do, but I don't know what the calling code (roundabouts, traffic lights, etc) is expecting from the iterator. So if I start filtering out some stuff, is that going to break code elsewhere? There was no documentation for exisitng code (that I could find at least), so difficult to know what it's really expecting.